activerecord/lib/arel/visitors/sqlite.rb

```diff
@@ -33,6 +33,31 @@ def visit_Arel_Nodes_IsDistinctFrom(o, collector)
           collector << " IS NOT "
           visit o.right, collector
         end
+
+        # Queries used in UNION should not be wrapped by parentheses,
+        # because it is an invalid syntax in SQLite.
+        def infix_value_with_paren(o, collector, value, suppress_parens = false)
```

Comment:
Need this for sqlite to make sure that queries used in `UNION` are not wrapped by parenthesis, because it is an invalid syntax.

`select .. union select ..` is good, but `(select ..) union (select ..)` is invalid 🤦